lib/db.js

```diff
@@ -0,0 +1,38 @@
+const MongoClient = require('mongodb').MongoClient,
+      assert = require('assert');
+
+class DB {
+    constructor(url) {
+        this.connected = false;
+
+        MongoClient.connect(url, (err, db) => {
+            assert.equal(null, err);
+            console.log("Connected to the database");
+
+            this.db = db;
+            this.connected = true;
+        });
+    }
+
+    insertItemData(itemdata) {
+        if (this.connected) {
+            this.db.collection('marketfloats').insert(JSON.parse(JSON.stringify(itemdata)));
```

Comment:
I was also encountering some very weird issues with the insert. AFAIK, there was a hidden property `$add` on the obj that was conflicting during the insert. Doing this removes whatever unwanted properties but it is certainly weird.